### DISCUSSION TITLE
[C128] (signed char) not supported?

### DISCUSSION THREAD

**Discussion by RetrofanC** [cc65/cc65#1951] [Q&A] [answered]
Hi Community, 

Not sure if this question belongs here but I give it a try. 
If it is considered basic C understanding please just mention and discard. 

Before porting my project to the C128 I have it running on a PC with CODE::BLOCKS.
As mentioned in my discussions earlier most is working very nicely but I have a snag;

In my code I am using; 

#define GetBYTE_pp(a)	( memory_read( cpu, ( a )++ & 0xffff ) )
.....

adr = IY + (signed char) GetBYTE_pp(PC);
PutBYTE(adr, GetBYTE_pp(PC));
 
 to change an 'unsigned char' into a 'signed char'.
 (To make a relative jump in the code forward or backward)  

However, with the CC65 tool chain this doesn't seem to work for some reason. 
It doesn't makes it a 'signed' char it just remains an 'unsigned char'. 

When I try this below it obviously doesn't work; 
 
 adr = IY + (*(signed char*) &GetBYTE_pp(PC));

To solve this it seems I have to introduce an extra variable and use a pointer re-declaration 
(not sure if this is the right word but you get the idea.....) 	
	
TempGetBYTE = GetBYTE_pp(PC);
adr = IY + (*((signed char*) &TempGetBYTE));
PutBYTE(adr, GetBYTE_pp(PC));

Now as for my questions to the community; 
       
Q1: Why doesn't CC65 works with ' (signed char) GetBYTE_pp(PC) ' or do I just have to write it slightly different? 
Q2: Does my option with introducing and extra variable and the pointer re-declaration do the same as the original line of code?
I think it should but somehow I got a feeling I might have a thinking error somewhere and they might not give the same result.

**Comment by mrdudz** [comment]
Did you try this with a recent built from github?

Also please provide a minimal, but complete, piece of code that demonstrates the problem

**Reply by RetrofanC** [reply, marked as answer]
Hi mrdudz & community, 

Sorry for not coming back on this in time to close it off. 

In the mean time and after some additional testing I found the following;  

**These are working as expected: ** 

(((int8_t) jump_value))

((signed int) jump_value)

{ unsigned char b;
b = jump_value;
return (*(signed char*) (&jump_value));
}

**This one isn't (and in hindsight shouldn't): ** 

(*(signed char*) (jump_value))

_(For completeness, why the first three didn't work in my code initially. 
Had to do with bank (memory) switching, all good now. )_
